packages/sanity/src/_internal/cli/threads/extractSchema.ts

```diff
@@ -0,0 +1,73 @@
+import {isMainThread, parentPort, workerData as _workerData} from 'node:worker_threads'
+
+import {extractSchema} from '@sanity/schema/_internal'
+import {type Workspace} from 'sanity'
+
+import {getStudioWorkspaces} from '../util/getStudioWorkspaces'
+import {mockBrowserEnvironment} from '../util/mockBrowserEnvironment'
+
+export interface ExtractSchemaWorkerData {
+  workDir: string
+  workspaceName?: string
+  enforceRequiredFields?: boolean
+}
+
+export interface ExtractSchemaWorkerResult {
+  schema: ReturnType<typeof extractSchema>
+}
+
+if (isMainThread || !parentPort) {
+  throw new Error('This module must be run as a worker thread')
+}
+
+const opts = _workerData as ExtractSchemaWorkerData
+const cleanup = mockBrowserEnvironment(opts.workDir)
+
+async function main() {
+  try {
+    const workspaces = await getStudioWorkspaces({basePath: opts.workDir})
+
+    const workspace = getWorkspace({workspaces, workspaceName: opts.workspaceName})
+
+    const {types} = workspace.schema._original || {types: []}
```

Comment:
If you're looking into using the compiled schema, you can use `workspace.schema` instead of `workspace.schema._original` here